api-spec/sftt/api-spec.md

```diff
@@ -205,14 +206,84 @@ Invite someone to join a team. Will send an email to all specified people that i
 }
 ```
 
-## `POST /teams/:team_id/join`
+## `POST /teams/:team_id/apply`
 
-Join this team. Any member that is not currently a part of a team can join any team.
+Apply to join this team. Any member can apply to join a team that they are not currently on. They will have to be approved by the team leader before becoming an actual member of the team.
 
 ### Request
 
 No request body.
 
+## `GET /teams/:team_id/applicants`
+
+Team Leader only.
+
+Get the info for anyone that has requested to join this team in a list.
+
+### Responses
+
+##### `200 OK`
+
+```json
+{
+  "applicants": [
+    {
+      "userId": INT,
+      "username": STRING,
+      "firstName": STRING,
+      "lastName": STRING,
```

Comment:
Doesn't matter if we return more info, but realistically we'd only show the leader their username when they apply, and we need the userID to accept or reject them but that's it.

Reply:
That's a good point, the real implementation doesn't use the first / last name so I'm going to scrap it